tools/cask/main.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2019 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// The package provides a transparent pass-through
+// for the Custodian CLI to a Custodian Docker Image
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/stdcopy"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+const CONTAINER_HOME string = "/home/custodian/"
+const IMAGE_NAME string = "cloudcustodian/c7n:latest"
+
+func main() {
+	fmt.Printf("Custodian Cask (%v)\n", IMAGE_NAME)
+
+	ctx := context.Background()
+
+	// Create a docker client
+	dockerClient := GetClient()
+
+	// Ensure latest docker image
+	Pull("docker.io/" + IMAGE_NAME, dockerClient, ctx)
```

Comment:
I have some minor misgivings about pulling every execution.. it adds about a second on a good network. wondering if we could do some sort of happy middle of just updating once a day, unless its forced/told to pull every execution or its if its on a release tag (not latest).. ie..  poking with the docker python client something like

```python
>>> import docker
>>> from datetime import datetime
>>> client = docker.from_env()
>>> image = client.images.get('cloudcustodian/c7n:latest')
>>> image.short_id
'sha256:fd87b364dd'
>>> delta = datetime.utcnow() - datetime.utcfromtimestamp(image.history()[-1]['Created'])
>>> delta
datetime.timedelta(days=8, seconds=69208, microseconds=564112)
>>> if delta.days > 0: 
... client.images.pull('cloudcustodian/c7n:latest')
```

Reply:
its actually only about .4 seconds vs docker naked, some timing numbers

```
(custodian) kapilt@realms-scythe ~/p/release-custodian> time docker run gcr.io/cloud-custodian/cli:latest version
0.8.44.1
        1.94 real         0.03 user         0.01 sys
(custodian) kapilt@realms-scythe ~/p/release-custodian> time custodian-cask version
Custodian Cask (cloudcustodian/c7n:latest)
latest: Pulling from cloudcustodian/c7n
Digest: sha256:3e9bd68e55dec29c5a7f31e38f0377c65544c63d72f546b1b10f8493369b8eda
Status: Image is up to date for cloudcustodian/c7n:latest
0.8.44.1
        2.35 real         0.00 user         0.00 sys
(custodian) kapilt@realms-scythe ~/p/release-custodian> time custodian version
0.8.44.0
        0.72 real         0.60 user         0.10 sys
```

Reply:
Yeah I had actually intended to just do something simple like write a marker file to `os.TempDir()` which I'd use to only do the pull once per hour - but when I saw how short the delay was I figured I'd worry about it later as part of a more robust set of options for selecting/managing which image to use.

But... since you had the same concern maybe I should just add it in now.